src/metabase/metabot.clj

```diff
@@ -95,16 +94,14 @@
   ([]
    "Show which card? Give me a part of a card name or its ID and I can show it to you. If you don't know which card you want, try `metabot list`.")
   ([card-id-or-name & _]
-   (let-404 [{card-id :id} (id-or-name->card card-id-or-name)]
-     (do-async (pulses/send-pulse! {:cards    [{:id card-id}]
-                                    :channels [{:channel_type   "slack"
-                                                :recipients     []
-                                                :details        {:channel *channel-id*}
-                                                :schedule_type  "hourly"
-                                                :schedule_day   "mon"
-                                                :schedule_hour  8
-                                                :schedule_frame "first"}]})))
-   "Ok, just a second..."))
+   (if-let [{card-id :id} (id-or-name->card card-id-or-name)]
+     (do
+       (do-async (let [attachments (pulse/create-and-upload-slack-attachments! [(pulse/execute-card card-id)])]
+                   (slack/post-chat-message! *channel-id*
+                                             nil
+                                             attachments)))
+       "Ok, just a second...")
+     (throw (Exception. "Not Found")))))
```

Comment:
`let-404` is useful for API responses, but it's useful in other places too. There are lots of places in our code where the `let-404` pattern is useful. We should definitely use it when it makes the code clearer and more succinct. It's perfect for this situation so we should use it instead of repeating the boilerplate